unity-renderer/Assets/Scripts/MainScripts/DCL/Components/CameraModeArea/CameraModeArea.cs

```diff
@@ -0,0 +1,176 @@
+            }
+        }
+
+        private static CameraModeAreasController areasController { get; } = new CameraModeAreasController();
+
+        private Collider playerCollider;
+        internal IUpdateEventHandler updateEventHandler;
+        internal int validCameraModes = 1 << (int)CameraMode.ModeId.FirstPerson | 1 << (int)CameraMode.ModeId.ThirdPerson;
+
+        internal bool isPlayerInside = false;
+
+        public Model areaModel { private set; get; } = new Model();
+        public IParcelScene areaScene { private set; get; }
+        public IDCLEntity areaEntity { private set; get; }
+
+        CameraMode.ModeId ICameraModeArea.cameraMode => areaModel.cameraMode;
+
+        IDCLEntity IEntityComponent.entity => areaEntity;
+
+        IParcelScene IComponent.scene => areaScene;
+
+        string IComponent.componentName => "CameraModeArea";
+
+        void IComponent.UpdateFromJSON(string json)
+        {
+            OnModelUpdated(areaModel.GetDataFromJSON(json) as Model);
+        }
+
+        void IComponent.UpdateFromModel(BaseModel model)
+        {
+            OnModelUpdated(model as Model);
+        }
+
+        IEnumerator IComponent.ApplyChanges(BaseModel model) => null;
+
+        void IComponent.RaiseOnAppliedChanges() { }
+
+        bool IComponent.IsValid() => true;
+
+        BaseModel IComponent.GetModel() => areaModel;
+
+        int IComponent.GetClassId() => (int)CLASS_ID_COMPONENT.CAMERA_MODE_AREA;
+
+        Transform IMonoBehaviour.GetTransform() => areaEntity?.gameObject.transform;
+
+        void ICleanable.Cleanup()
+        {
+            Dispose();
+        }
+
+        void IEntityComponent.Initialize(IParcelScene scene, IDCLEntity entity)
+        {
+            Initialize(scene, entity, Environment.i.platform.updateEventHandler, DataStore.i.player.playerCollider.Get());
+        }
+
+        internal void OnModelUpdated(in Model newModel)
+        {
+            if (!IsValidCameraMode(newModel.cameraMode))
+            {
+                newModel.cameraMode = CommonScriptableObjects.cameraMode.Get();
+            }
+
+            bool cameraModeChanged = newModel.cameraMode != areaModel.cameraMode;
+            areaModel = newModel;
+
+            if (cameraModeChanged && isPlayerInside)
+            {
+                areasController.ChangeAreaMode(this, areaModel.cameraMode);
+            }
+        }
+
+        internal void Initialize(in IParcelScene scene, in IDCLEntity entity, in IUpdateEventHandler updateEventHandler, in Collider playerCollider)
+        {
+            areaScene = scene;
+            areaEntity = entity;
+            this.updateEventHandler = updateEventHandler;
+            this.playerCollider = playerCollider;
+
+            updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, Update);
+        }
+
+        internal void Update()
+        {
+            bool playerInside = IsPlayerInsideArea();
+
+            switch (playerInside)
```

Comment:
First time I see this `switch / when` thing. Why is needed?

Reply:
It's the same as:
```
if (playerInside && !isPlayerInside)
{
    // ...
}
else if (!playerInside && isPlayerInside)
{
    // ...
}
```
but way more cool and easier to read (was a rider's suggestion) :P

Reply:
Aahhh the naming confused me, I didnt notice `playerInside` is different than `isPlayerInside`.

Reply:
Interesting, I too didn't know we could have multiple conditions in the switch cases this way, nice